test.c

```diff
@@ -49,15 +49,28 @@ int main() {
   ok = hasUniqueChars(string3);
   assert(ok);
   
-  // TODO: add your tests here
+  // Test 3: Unique characters should return true
+  strcpy(string3, "abcdefg");
+  ok = hasUniqueChars(string3);
+  assert(ok);
+
+  // Test 4: A string with spaces but unique characters should still return true
+  strcpy(string3, "a b c d e f g");
+  ok = hasUniqueChars(string3);
+  assert(ok);
+
+  // Test 5: A string with a newline (should fail)
+  // strcpy(string3, "valid string\n");
+  // ok = hasUniqueChars(string3);
+  // assert(!(ok));
 
```

Comment:
Missing many many tests, you never test the other set of characters, or capitals, or...